### ISSUE TITLE
Reward - intrinsic reward or extrinsic reward?

### ISSUE THREAD

**Issue by wechto** [Steven-Ho/VALOR#1]
Hello, I am a student who is just beginning to learn RL, I have run the examples of VALOR code. 

I have a question about the reward, which is gotten from the extrinsic environment. I think the reward should come from equation (2) [https://arxiv.org/pdf/1807.10299.pdf](url). Following the similar operation such as VIC and  DIAYN, I think the reward should follow the equation:

> $r_t = log q_{\phi}(z|s_{t}) - log p(z)$
![image](https://user-images.githubusercontent.com/25589094/62552544-63ca6f00-b8a0-11e9-9bfd-950a83fb0e6a.png)

So my question is that why the code does not use the same intrinsic reward?

thank you in advance.

**Comment by Steven-Ho** [comment]
Sorry for delay. The intrinsic reward was added on policy loss through
`pi_loss = -(logp*(k*adv+pos)).mean()`
The `pos` term represents log q(z|\tau).

**Comment by Steven-Ho** [comment]
@O151 
And the reward in VALOR is not the same as VIC. VALOR calculates the log possibility conditioned on the whole trajectory and evenly adds it as intrinsic reward on every step. The -logp(z) term is a constant since the prior is uniform. So it's not necessary in reward.

**Comment by AlexandraFilimokhina** [comment]
Hello, Steven-Ho! 
I am a novice in RL and trying to understand VALOR for my project. 
I have some questions about your implementation of the algorithm. At first, can you elaborate on scipy.signal.lfilter, which you use in buffer.py. What do you use it for?
Secondly, how do you compute the first part of the policy loss: pi_loss = -(logp*(k*adv+pos)).mean(), I keep in mind k*adv. What is k and why is it equal to 1?
Thank you in advance.
